File: conversions/django_models/trait.py
from django.db import models
from django.core.validators import MaxValueValidator
from .abstract_element_model import AbstractElementModel

class Trait(AbstractElementModel):

    # Qualitative
    social_effects = models.TextField(blank=True, null=True)
    physical_effects = models.TextField(blank=True, null=True)
    skill_effects = models.TextField(blank=True, null=True)
    personality_effects = models.TextField(blank=True, null=True)
    artistic_effects = models.TextField(blank=True, null=True)
    behaviour_effects = models.TextField(blank=True, null=True)

    # Quantitative
    charisma = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    coercion = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    capability = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    compassion = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    creativity = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    courage = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)

    # World
    anti_trait = models.ForeignKey("Trait", on_delete=models.SET_NULL, blank=True, null=True, related_name="trait_anti_trait")
    empowered_abilities = models.ManyToManyField("Ability", blank=True, related_name="trait_empowered_abilities")

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Trait.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Trait : Element
{
    [JsonProperty("social_effects"), TextAttribute("")]
    public string socialEffects;
    [JsonProperty("physical_effects"), TextAttribute("")]
    public string physicalEffects;
    [JsonProperty("functional_effects"), TextAttribute("")]
    public string functionalEffects;
    [JsonProperty("personality_effects"), TextAttribute("")]
    public string personalityEffects;
    [JsonProperty("behaviour_effects"), TextAttribute("")]
    public string behaviourEffects;
    [JsonProperty("charisma"), Integer(100)]
    public int charisma;
    [JsonProperty("coercion"), Integer(100)]
    public int coercion;
    [JsonProperty("competence"), Integer(100)]
    public int competence;
    [JsonProperty("compassion"), Integer(100)]
    public int compassion;
    [JsonProperty("creativity"), Integer(100)]
    public int creativity;
    [JsonProperty("courage"), Integer(100)]
    public int courage;
    [JsonProperty("significance"), TextAttribute("")]
    public string significance;
    [JsonProperty("anti_trait"), ReferenceAttribute(typeof(Trait))]
    public string antiTrait;
    [JsonProperty("empowered_abilities"), ReferenceAttribute(typeof(Ability), true)]
    public string empoweredAbilities;
}
